pkg/detectors/privatekey/ssh.go

```diff
@@ -75,13 +75,24 @@ func firstResponseFromSSH(ctx context.Context, parsedKey any, username, hostport
 	var output bytes.Buffer
 	session.Stderr = &output
 
-	err = session.Shell()
-	if err != nil {
+	if err = session.Shell(); err != nil {
 		return "", err
 	}
-	_ = session.Wait()
 
-	return output.String(), err
+	// Wait for the session to complete or context cancellation.
+	// If we don't check for context cancellation, the session will hang indefinitely.
+	done := make(chan error, 1)
+	go func() {
+		done <- session.Wait()
+	}()
```

Comment:
If the session hangs indefinitely, would this be a goroutine leak?

Reply:
If the session hangs indefinitely, the `ctx` timeout should capture it. The case will then close the session, preventing a goroutine leak—at least, that’s how I believe it should work.

Reply:
Could we add a test for that? :eyes: